apis/multiarch/common/plugins/base_plugins_test.go

```diff
@@ -0,0 +1,358 @@
+package plugins_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/openshift/multiarch-tuning-operator/apis/multiarch/common/plugins"
+	"github.com/openshift/multiarch-tuning-operator/apis/multiarch/v1alpha1"
+	"github.com/openshift/multiarch-tuning-operator/apis/multiarch/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestBasePlugin_IsEnabled(t *testing.T) {
+	tests := []struct {
+		name     string
+		enabled  bool
+		expected bool
+	}{
+		{"Enabled Plugin", true, true},
+		{"Disabled Plugin", false, false},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			plugin := &plugins.BasePlugin{Enabled: tt.enabled}
+			if plugin.IsEnabled() != tt.expected {
+				t.Errorf("Expected IsEnabled() to be %v, got %v", tt.expected, plugin.IsEnabled())
+			}
+		})
+	}
+}
+
+func TestBasePlugin_Name(t *testing.T) {
+	plugin := &plugins.BasePlugin{}
+	if plugin.PluginName() != "BasePlugin" {
+		t.Errorf("Expected Name() to return 'BasePlugin', got %s", plugin.PluginName())
+	}
+}
+
+func TestNodeAffinityScoring_Validation(t *testing.T) {
+	tests := []struct {
+		name       string
+		platforms  []plugins.NodeAffinityScoringPlatformTerm
+		shouldFail bool
+	}{
+		{
+			name: "Valid Platforms",
+			platforms: []plugins.NodeAffinityScoringPlatformTerm{
+				{Architecture: "ppc64le", Weight: 50},
+				{Architecture: "amd64", Weight: 30},
+			},
+			shouldFail: false,
+		},
+		{
+			name: "Empty Platforms",
+			platforms: []plugins.NodeAffinityScoringPlatformTerm{
+				{Architecture: " ", Weight: 10},
```

Comment:
```suggestion
				{Architecture: "", Weight: 10},
```

empty string or delete in favor of the next one ('invalid_arch')